corehq/apps/appstore/management/commands/export_exchange.py

```diff
@@ -0,0 +1,54 @@
+import csv
+
+from datetime import datetime
+from django.core.management.base import BaseCommand
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models import CouchUser
+from dimagi.utils.parsing import ISO_DATETIME_FORMAT
+
+
+class Command(BaseCommand):
+    help = """
+        Pulls data on domain snapshots (projects in the late CommCare Exchange).
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('outfile')
+
+    def handle(self, outfile, **options):
+        snapshots = Domain.get_db().view(
+            'domain/snapshots',
+            startkey=[],
+            endkey=[{}],
+            reduce=False,
+            include_docs=True,
+        ).all()
+        snapshots = [s['doc'] for s in snapshots]
+        snapshots = [s for s in snapshots if s['published'] and s['is_snapshot'] and s.get('snapshot_head')]
+        rows = []
+        for s in snapshots:
+            domain = Domain.wrap(s)
+            user = None
+            if s['cda']['user_id']:
+                user = CouchUser.get_by_user_id(s['cda']['user_id'])
+            author = s['author']
+            if not author and user:
+                author = user.human_friendly_name
+            row = (
+                domain.copied_from,
+                s['title'],
+                author,
+                s['short_description'],
+                s['project_type'],
+                user.username if user else None,
+                datetime.strptime(s['snapshot_time'], ISO_DATETIME_FORMAT).strftime('%b %d, %Y'),
+                ', '.join(set([lang for app in domain.applications() for lang in app.langs])),
+                s['license'],
+            )
+            rows.append(row)
+
+        with open(outfile, 'w', encoding='utf-8') as out:
```

Comment:
(non blocking)
i am not sure how many snapshots you are going to iterate here. I always find it useful to have some sort of tracking in place to know how things are going, so i have used `with_progress_bar` for such commands.
Also writing along with iteration itself also helps see what information is being put while things are being iterated. This is helpful to terminate earlier if the information is not being put correctly.

Reply:
Agreed in general. There are only about 600 of these, so this runs quickly.